generators/entity-server/templates/src/test/java/package/repository/search/_EntitySearchRepositoryMockConfiguration.java

```diff
@@ -0,0 +1,34 @@
+<%#
+ Copyright 2013-2018 the original author or authors from the JHipster project.
+
+ This file is part of the JHipster project, see http://www.jhipster.tech/
+ for more information.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+-%>
+package <%=packageName%>.repository.search;
+
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * Configure a Mock version of <%= entityClass %>SearchRepository to test the
+ * application without starting Elasticsearch.
+ */
+@Configuration
+public class <%=entityClass%>SearchRepositoryMockConfiguration {
```

Comment:
instead of generating one per entity cant we use the same for all as they only change in name?

Reply:
Not sure if it's doable... You also sometimes only Mock one, and maybe some people will want something more flexible as they will do stuff in their Mock... I'm not sure of what we can/should do here, I agree it's quite a lot of files.

Reply:
since the class doesn't use anything specific to an entity, I think we can generate this during project creation in test package and use it for as the Search repo for all entity and if someone wants a specific mock they can always create a new one or extend this right? otherwise it will be too many of these

Reply:
I'm not sure of how you want to do this - just give it a try and show me, unless it's complicated

Reply:
ok sure, i'll update it in another PR after merging this